src/main/scala/it/pagopa/interop/catalogprocess/service/impl/PartyManagementServiceImpl.scala

```diff
@@ -22,19 +23,19 @@ final case class PartyManagementServiceImpl(invoker: PartyManagementInvoker, api
     Logger.takingImplicit[ContextFieldsToLog](this.getClass)
 
   override def getInstitution(
-    id: UUID
-  )(implicit contexts: Seq[(String, String)], ec: ExecutionContext): Future[Institution] = for {
-    uid <- getUidFuture(contexts)
-    request = api.getInstitutionById(id)(uid)
-    result <- invoker.invoke(request, s"Institution retrieved for id ${id.toString}")
-  } yield result
+    id: String
+  )(implicit contexts: Seq[(String, String)], ec: ExecutionContext): Future[Institution] = withUid(uid =>
+    id.toFutureUUID.flatMap { idAsUUID =>
+      val request = api.getInstitutionById(idAsUUID)(uid)
+      invoker.invoke(request, s"Institution retrieved for id ${id.toString}")
+    }
+  )
 
   override def getBulkInstitutions(
-    identifiers: BulkPartiesSeed
-  )(implicit contexts: Seq[(String, String)], ec: ExecutionContext): Future[BulkInstitutions] = for {
-    uid <- getUidFuture(contexts)
-    request = api.bulkInstitutions(identifiers)(uid)
-    result <- invoker
-      .invoke(request, s"Institutions retrieved for identifiers ${identifiers.partyIdentifiers.mkString(",")}")
-  } yield result
+    identifiers: List[String]
+  )(implicit contexts: Seq[(String, String)], ec: ExecutionContext): Future[BulkInstitutions] = withUid { uid =>
+    val ids     = BulkPartiesSeed(identifiers.map(UUID.fromString))
```

Comment:
⚠️ `UUID.fromString` can throw an exception.
Why not using `toFutureUUID` here?